marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java

```diff
@@ -71,8 +72,12 @@
     @ResponseBody
     @ApiOperation(value = "Update a flow", response = FlowInfo.class)
     @Secured("ROLE_writeFlow")
-    public ResponseEntity<JsonNode> updateFlowInfo(@PathVariable String flowName, @RequestBody UpdateFlowInfo info) {
-        return ResponseEntity.ok(newFlowService().updateFlowInfo(flowName, info.description));
+    public ResponseEntity<JsonNode> updateFlow(@PathVariable String flowName, @RequestBody JsonNode flow) {
+
+        ArrayNode stepsArray = (ArrayNode) flow.get("steps");
+        String flowDesc = flow.get("description").asText();
+
+        return ResponseEntity.ok(newFlowService().updateFlow(flowName, flowDesc, stepsArray));
```

Comment:
I think you can pass the updatedFlow payload from the UI directly to the endpoint and do all the processing there. The "updatedFlow" endpoint will receive a JsonNode as input.

Reply:
@devinnicholson 
Can you fix this as well ?